pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -58,8 +58,8 @@ func (s *Service) ReconcileLoadBalancer(openStackCluster *infrav1.OpenStackClust
 	if err != nil {
 		return err
 	}
-	if lb.ProvisioningStatus != loadBalancerProvisioningStatusActive {
-		return fmt.Errorf("load balancer %q is not in expected state %s, current state is %s", lb.ID, loadBalancerProvisioningStatusActive, lb.ProvisioningStatus)
+	if err := s.waitForLoadBalancerActive(lb.ID); err != nil {
```

Comment:
does this means if a LB is not gonna success due to some issue
then for every reconcile we will wait for it to be ACTIVE but we actually know openstack has no reconcile mechanism
so will it reasonable to just wait for `pending create` status LB only?

Reply:
I also stumbled across this.
Currently we are waiting for the LB to be active on each reconciliation. IMHO requeuing would be the better way.

Reply:
Besides `PENDING_CREATE`, there is also `PENDING_UPDATE` (which triggers #1197) and `PENDING_DELETE`: [Octavia Provisioning Status Codes](https://docs.openstack.org/api-ref/load-balancer/v2/index.html#status-codes). I think we want to wait for all of those.

Moving the `waitForLoadBalancerActive` into `ReconcileLoadBalancer` was the "minimal change" that fixed the bug in a clean way without changing the logic in any major way. In general I think we should move everything to requeuing.

Reply:
@jichenjc One option would be to abort and return in `waitForLoadBalancerActive` if the loadbalancer is in a fatal state (`DELETED`, `ERROR`). Would that be fine for you? 

@tobiasgiese In most cases we do not need to wait because the LB is already `ACTIVE` and `waitForLoadBalancerActive` returns immediatly. Do you still want me to implement requeuing in this PR?

Reply:
@apricote like @mdbooth also wrote, I'm totally fine with merging your PR as is and implement the requeuing in a follow-up PR 👍🏻